### DISCUSSION TITLE
No 'Access-Control-Allow-Origin' error when redirecting the frontend

### DISCUSSION THREAD

**Discussion by tempcoderr** [fastapi/fastapi#6046] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
app.add_middleware(
  CORSMiddleware,
  allow_origins=['http://localhost:3000', 'http://localhost:5173', 'http://localhost:3001'],
  allow_credentials=True,
  allow_methods=["GET", "POST", "PUT", "HEAD", "OPTIONS"],
  allow_headers=["Content-Type", "Set-Cookie", "Access-Control-Allow-Headers", 'Content-Type', 'Authorization', 'Access-Control-Allow-Origin'],
)

@router.put("/provider-data")
def temp(data: dict, Auth_Token: Optional[str] = Cookie(None)):
    if Auth_Token is None:
        return False
    user = add_provider_user_data(data["username"], data["password"], Auth_Token)
    if not user:
        return False
    return RedirectResponse(url='http://localhost:3000/test')
```

### Description

I have a FastAPI backend (port: 8000) and NextJs frontend (port: 3000) and from the frontend I am sending a put request which inserts user data into a user's object, and after it is done send a redirection response to `http://localhost:3000/test` (redirect to a frontend path), but it is not doing so and giving CORS No 'Access-Control-Allow-Origin' error.

### Operating System

Windows

### FastAPI Version

0.88.0

### Python Version

3.10.8

### Additional Context

My frontend code is:
```javascript
  const handleSubmit = () => {
    axios
      .put(
        "http://localhost:8000/provider-data",
        { username, password },
        {
          headers: {
            "Access-Control-Allow-Origin": "http://localhost:3000",
          },
          withCredentials: true,
        }
      )
      .then((res) => {
        console.log(res);
        setUsername("");
        setPassword("");
      })
      .catch((err) => {
        console.log(err);
      });
  };
```

**Comment by Lenclove** [comment]
@tempcoderr I think `RedirectResponse` doesn't return a `Access-Control-Expose-Headers` response header. I read the `RedirectResponse` implementation of `starlette`. It doesn't contain the `Access-Control-Expose-Headers` header. I think you should add these headers by yourself. About the CORS, you can refer to this [link](https://developer.mozilla.org/zh-CN/docs/Web/HTTP/CORS).

And `RedirectResponse` also has a keyword argument -- `header=...`, you can see the starlette's docs and source code.

**Reply by BerkKilicoglu** [reply]
I tried this but it did not solve the problem is there an update?
